File: string/detect-capital/detect-capital.go
package detectcapital

func detectCapitalUse(word string) bool {
	nOfCaps := 0
	for i := 0; i < len(word); i++ {
		if word[i] < 'a' {
			nOfCaps++
		}
	}

	return (nOfCaps == len(word)) || (nOfCaps == 1 && word[0] < 'a') || (nOfCaps == 0)
}

func detectCapitalUseComplex(word string) bool {
	if word[0] < 'a' {
		nOfCaps := 0
		for i := 1; i < len(word); i++ {
			if word[i] < 'a' {
				nOfCaps++
			}
		}
		return nOfCaps == (len(word)-1) || nOfCaps == 0
	}

	for i := 1; i < len(word); i++ {
		if word[i] < 'a' {
			return false
		}
	}

	return true
}

File: string/detect-capital/detect-capital.ts
function detectCapitalUse(word: string): boolean {
    let nOfCaps = 0;

    for (let c of word) {
        if (c === c.toUpperCase()) {
            nOfCaps++;
        }
    }

    return (nOfCaps === 0) ||
        (word[0] === word[0].toLocaleUpperCase() && nOfCaps === 1) ||
        (nOfCaps === word.length)
};
